src/core/src/test/java/org/apache/jmeter/gui/util/JSyntaxTextAreaTest.java

```diff
@@ -41,7 +45,37 @@ public void testSetLanguage() {
             textArea.setLanguage(null);
             assertEquals(SyntaxConstants.SYNTAX_STYLE_NONE, textArea.getSyntaxEditingStyle());
         } catch (HeadlessException he) {
-            // Does not work in headless mode
+            // Does not work in headless mode, which depends on value of java.awt.headless property
+            // and the OS (e.g. might work on MacOS and not on Linux due to missing X11).
```

Comment:
I don't understand this new comment. Can you explain it to me?

Reply:
How I got here: I have Kotlin code (to generate a jmx file based on open-api spec) that calls setText() and getText() for the json payload. When running from Kotlin code on MacOS laptop, the code runs fine in Intellij: getText() returns a value. When running it from Kotlin _script_, the getText() returns null after setText(). Why? Debugging shows that `java.awt.headless=true` when running Kotlin _script_. When running Kotlin code the `java.awt.headless` property is not present, so does not run headless. (In that case it I guess it might work because `awt.toolkit=sun.lwawt.macosx.LWCToolkit` is present). Then on linux docker (target system) running non-headless does not work because it says no DISPLAY is set (awt.toolkit is missing?). And running headless (with `java.awt.headless=true`), the getText() returns null.

Reply:
I've filed https://github.com/bobbylight/RSyntaxTextArea/issues/424 to `RSyntaxTextArea`. I hope we can remove headless workaround then.